components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java

```diff
@@ -111,6 +112,12 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
                     return authenticationResult;
                 }
 
+                // If the request is coming to me endpoint, store the token id to the thread local.
+                if (authenticationRequest.getRequest() != null && authenticationRequest.getRequest().getRequestURI()
+                        .toLowerCase().endsWith(SCIM_ME_ENDPOINT_URI) && accessToken != null) {
```

Comment:
can't authenticationRequest.getRequest().getRequestURI() be null?

Reply:
Updated